common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -368,4 +381,70 @@ private static void closeQuietly(Closeable c) {
     private NativeLibraryLoader() {
         // Utility
     }
+
+    private static final class NoexecFileStoreDetector {
+        static final Method TO_PATH_METHOD;
+        static final Method GET_FILE_STORE_METHOD;
+        static final Method MOUNT_ENTRY_METHOD;
+        static final Method HAS_OPTION_METHOD;
+
+        static {
+            Method toPathMethod = null;
+            Method getFileStoreMethod = null;
+            Method mountEntryMethod = null;
+            Method hasOptionMethod = null;
+            try {
+                // Use reflection so Java 6 fails gracefully.
+                toPathMethod = File.class.getMethod("toPath");
+                Class<?> pathClass = Class.forName("java.nio.file.Path");
```

Comment:
I would prefer if we explicitly reference the type instead of relying upon implicit string/reflection lookup:

```java
if (PlatformDependent.javaVersion() >= 7) {
// reference java.nio.file.Path directly
} else {
  MOUNT_ENTRY_METHOD = null;
  ...
}
```

You will also have to add an expiation to the `pom.xml` to allow this jdk7+ API to be referenced.